codecs/avif/Makefile

```diff
@@ -0,0 +1,76 @@
+ROOT_DIR = node_modules
+
+CODEC_DIR = $(ROOT_DIR)/libavif
+CODEC_BUILD_DIR = build
+CODEC_OUT := $(CODEC_DIR)/$(CODEC_BUILD_DIR)/libavif.a 
+
+CODEC_URL = "https://github.com/AOMediaCodec/libavif"
+CODEC_VERSION = "v0.8.0"
+
+LIBAOM_DIR := $(CODEC_DIR)/ext/aom
+LIBAOM_BUILD_DIR = build.libavif
+LIBAOM_OUT := $(LIBAOM_DIR)/$(LIBAOM_BUILD_DIR)/libaom.a
+
+LIBAOM_URL = "https://aomedia.googlesource.com/aom/"
+LIBAOM_VERSION = "v2.0.0"
+
+OUT_JS = enc/avif_enc.js dec/avif_dec.js
+OUT_WASM = $(OUT_JS:.js=.wasm)
+
+.PHONY: all clean
+
+all: $(OUT_JS)
+
+%.js: %.cpp $(LIBAOM_OUT) $(CODEC_OUT)
+	$(CXX) \
+		-I $(CODEC_DIR)/include \
+		${CXXFLAGS} \
+		${LDFLAGS} \
+		--bind \
+		--closure 1 \
+		-s ALLOW_MEMORY_GROWTH=1 \
+		-s MODULARIZE=1 \ -s 'EXPORT_NAME="$(basename $(@F))"' \
+		-o $@ \
+		$+
+
+$(LIBAOM_OUT): $(LIBAOM_DIR)/README.md
+	mkdir -p $(LIBAOM_DIR)/$(LIBAOM_BUILD_DIR) && \
+	cd $(LIBAOM_DIR)/$(LIBAOM_BUILD_DIR) && \
+	emcmake cmake \
+		-DCMAKE_BUILD_TYPE=Release \
+		-DENABLE_CCACHE=0 \
+		-DAOM_TARGET_CPU=generic \
+		-DENABLE_DOCS=0 \
+		-DENABLE_TESTS=0 \
+		-DCONFIG_ACCOUNTING=1 \
+		-DCONFIG_INSPECTION=0 \
+		-DCONFIG_MULTITHREAD=0 \
+		-DCONFIG_RUNTIME_CPU_DETECT=0 \
+		-DCONFIG_WEBM_IO=0 \
+		../ && \
+	$(MAKE) 
+
+$(LIBAOM_DIR)/README.md: $(CODEC_DIR)/README.md
+	mkdir -p $(LIBAOM_DIR) && \
+	cd $(LIBAOM_DIR) && \
+	curl -L $(LIBAOM_URL)/+archive/$(LIBAOM_VERSION).tar.gz | tar -xzf -
+
+$(CODEC_OUT): $(LIBAOM_OUT)
+	mkdir -p $(CODEC_DIR)/$(CODEC_BUILD_DIR) && \
+	cd $(CODEC_DIR)/$(CODEC_BUILD_DIR) && \
+	emcmake cmake \
+		DCMAKE_BUILD_TYPE=Release \
+		-DAVIF_CODEC_AOM=1 \
```

Comment:
Does this need similar `-DENABLE_DOCS=0` and so on or are they disabled by default here?

Reply:
Don’t _think_ so, I was following the docs in the README

Reply:
You just mentioned earlier you had to build a bunch of examples, and noticed that they don't seem to be disabled in either of the codecs configs.

Reply:
Ah no, the examples are in libaom and still get built :S

Reply:
Looks like there is indeed `ENABLE_EXAMPLES`, in the spirit of other similar options, that you can disable: https://aomedia.googlesource.com/aom/+/refs/heads/master/CMakeLists.txt#227